src/ClassDiscovery.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Http\Discovery;
+
+/**
+ * Registry that based find results on class existence
+ *
+ * @author David de Boer <david@ddeboer.nl>
+ */
+abstract class ClassDiscovery
+{
+    /**
+     * Cached class
+     *
+     * @var object
+     */
```

Comment:
To be clear: when a static property is declared in the parent, but not the child then setting it in the child will set it in the parent, since there is no context.